### PR TITLE
mobile: add new fields to claims and appeals overview response

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
adds 3 fields to claims and appeals overview endpoint

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/8822

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/mobile/docs/openapi.yaml
Comment (aherzberg) [inline modules/mobile/docs/openapi.yaml:1]: created separate examples of a claim and appeal like we do in appointments to show that appeals will have nil for the new fields.

File: modules/mobile/app/models/mobile/v0/claim_overview.rb
Comment (aherzberg) [inline modules/mobile/app/models/mobile/v0/claim_overview.rb:1]: note that the ticket calls to add 4 fields and this is only 3 but the 4th one phaseChangeDate is already being used for the `updated at` field.

Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/claim_overview.rb:1]: I've added a note in the docs that `updated at` is the latest phase change date as well

**Review by kpethtel — APPROVED** [review]
One small suggestion, but it looks good.

File: modules/mobile/app/models/mobile/v0/adapters/lighthouse_claims_overview.rb [hunk 1]
```diff
@@ -35,6 +38,10 @@ def date(attribute)
 
           Date.strptime(attribute, '%Y-%m-%d').iso8601
         end
+
+        def phase_to_number(phase)
```
[the remark this answers is not in the document]
Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/adapters/lighthouse_claims_overview.rb]: yeah, makes sense. added to the helper.

**Review by kpethtel — APPROVED** [review]
Looks good. 👍

**Pull request merged**